python/ml4ir/base/pipeline.py

```diff
@@ -381,28 +382,26 @@ def run(self):
                     pad_sequence=self.args.pad_sequence_at_inference,
                 )
 
+            # Write experiment details to experiment tracking dictionary
+            # Add command line script arguments
+            experiment_tracking_dict.update(vars(self.args))
+
+            # Add feature config information
+            experiment_tracking_dict.update(self.feature_config.get_hyperparameter_dict())
+
+            # Add train and test metrics
+            experiment_tracking_dict.update(train_metrics)
+            experiment_tracking_dict.update(test_metrics)
+
+            # Add optimizer and lr schedule
+            experiment_tracking_dict.update(relevance_model.model.optimizer.get_config())
+
         except Exception as e:
             self.logger.error("!!! Error Training Model: !!!\n{}".format(str(e)))
             traceback.print_exc()
             job_status = "_FAILURE"
             job_info = "{}\n{}".format(str(e), traceback.format_exc())
 
-        # Write experiment tracking data in job status file
-        experiment_tracking_dict = dict()
-
-        # Add command line script arguments
-        experiment_tracking_dict.update(vars(self.args))
-
-        # Add feature config information
-        experiment_tracking_dict.update(self.feature_config.get_hyperparameter_dict())
-
-        # Add train and test metrics
-        experiment_tracking_dict.update(train_metrics)
-        experiment_tracking_dict.update(test_metrics)
-
-        # Add optimizer and lr schedule
-        experiment_tracking_dict.update(relevance_model.model.optimizer.get_config())
-
         job_info = pd.DataFrame.from_dict(
```

Comment:
If there is an issue, `job_info` here overwrites the exception and put in its place an empty dict. Do you want to move it to line 398, in the try statement?

Reply:
Yeah. Let me do that.